tests/nexus_test_base.go

```diff
@@ -88,8 +89,9 @@ func (s *NexusTestBaseSuite) versionedNexusTaskPoller(ctx context.Context, taskQ
 			TaskToken: res.TaskToken,
 			Error:     handlerError,
 		})
-		// There's no clean way to propagate this error back to the test that's worthwhile. Panic is good enough.
-		if err != nil && ctx.Err() == nil {
+		// Ignore if context is already cancelled or if the task is not found.
+		if err != nil && ctx.Err() == nil && !errors.As(err, new(*serviceerror.NotFound)) {
```

Comment:
If the context was canceled, then `ctx.Err()` returns an "context canceled" error.

Reply:
Don't understand the comment. We want to panic only if the context wasn't canceled.

Reply:
Oh, right! I misread the code. Makes sense now.